### PR TITLE
Add suffixName on Form

### PR DESCRIPTION
**Pull request by flavienaudin**
Hi,
I detected an exception thrown when I have two "threads" on a page
"Serialization of 'Closure' is not allowed" in the file FormDataCollector of Symfony, when it's tryiing to generate the dev Profiler of Symfony.
I implemented a fix by adding a suffix on form name, when creating a form. I would like to share it and also get your opinion about the possibility of having two or more threads in a page.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,4 +4,4 @@ phpunit.xml
 Tests/autoload.php
 Tests/Functional/app/Behat/nginx.conf
 Tests/Functional/web/bundles
-vendor
+vendor
```
Comment (stof) [inline .gitignore]: removing the final EOL should be reverted

File: FormFactory/CommentFormFactoryInterface.php [hunk 2]
```diff
@@ -23,10 +23,11 @@
...
-     * @param mixed $method
-     * @param array $options
+     * @param mixed $data
+     * @param array $options => method
+     * @param string $name_suffix
      *
      * @return FormInterface
      */
-    public function createForm($data = null, $options = array());
+    public function createForm($data = null, $options = array(), $name_suffix = null);
```
Comment (stof) [inline FormFactory/CommentFormFactoryInterface.php]: this is a BC break

File: FormFactory/CommentableThreadFormFactoryInterface.php [hunk 3]
```diff
@@ -21,7 +21,8 @@
     /**
      * Creates a open thread form
      *
+     * @param $name_suffix string the suffix to name the form
      * @return FormInterface
      */
-    public function createForm();
+    public function createForm($name_suffix = null);
```
Comment (stof) [inline FormFactory/CommentableThreadFormFactoryInterface.php]: same here about being a BC break

File: FormFactory/CommentableThreadFormFactoryInterface.php [hunk 4]
```diff
@@ -21,7 +21,8 @@
     /**
      * Creates a open thread form
      *
+     * @param $name_suffix string the suffix to name the form
```
Comment (stof) [inline FormFactory/CommentableThreadFormFactoryInterface.php]: invalid phpdoc. The type comes before the variable name

File: FormFactory/CommentFormFactoryInterface.php (hunk 2 above)
Comment (flavienaudin) [inline FormFactory/CommentFormFactoryInterface.php]: I guess that the BC break (Backward Compatibility break, right ?) is on all interfaces ? but only on interfaces, not on the implementation classes ?
So, can I just remove the $name_suffix in the interface function prototype  ?

Sorry I'm new about making pull request and sharing code like this, but I'm glad to learn

thank for your quick feedback.

**Comment by XWB** [comment]
Closing as this PR is outdated.

**Pull request closed without merging**